src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/refdatafixes/ReferenceDataFixesService.java

```diff
@@ -119,4 +125,56 @@ public void initAdminData(AdminData adminData) {
         adminData.setExistingJudgeCode(null);
         adminData.setRequiredJudgeCode(null);
     }
+
+    public AdminData insertClaimServedDate(AdminDetails adminDetails,
+                                           String authToken,
+                                           RefDataFixesCcdDataSource dataSource,
+                                           List<String> errors) {
+        AdminData adminData = adminDetails.getCaseData();
+        String caseTypeId = getTribunalOffice(adminData.getTribunalOffice());
+        List<String> dates = getDateRangeForSearch(adminDetails, OLD_DATE_TIME_PATTERN);
+        String dateFrom = dates.get(0);
+        String dateTo = dates.get(1);
+        try {
+            List<SubmitEvent> submitEvents = dataSource.getDataForInsertClaimDate(caseTypeId, dateFrom, dateTo, ccdClient);
+            LocalDate claimServedDate;
+            if (CollectionUtils.isNotEmpty(submitEvents)) {
+                log.info("Cases Searched for inserting claim served date: " + submitEvents.size());
+                for (SubmitEvent submitEvent : submitEvents) {
+                    List<CaseEventDetail> caseEventDetails = ccdClient.retrieveCaseEventDetails(authToken,
+                            getTribunalOffice(adminData.getTribunalOffice()),
+                            adminDetails.getJurisdiction(), String.valueOf(submitEvent.getCaseId()));
+                    if (CollectionUtils.isNotEmpty(caseEventDetails)) {
+                        List<CaseEventDetail> generateCorrespondenceEvents = caseEventDetails
+                                .stream().filter(i -> i.getId().equals(
+                                        GENERATE_CORRESPONDENCE)).collect(Collectors.toList());
+                        if (CollectionUtils.isNotEmpty(generateCorrespondenceEvents)) {
+                            claimServedDate = generateCorrespondenceEvents.stream()
+                                    .sorted(CaseEventDetail::comparedTo)
+                                    .collect(Collectors.toList())
+                                    .get(0).getCreatedDate().toLocalDate();
+                            CaseData caseData = submitEvent.getCaseData();
+                            if (caseData.getClaimServedDate() == null) {
```

Comment:
6 levels of indentation. This is unreadable

Reply:
refactored the code now